documentation/proposals/Proposal - Generic Math.md

```diff
@@ -0,0 +1,1189 @@
+    static abstract TVector operator /(TVector left, TScalar right);
+
+    /// <summary>Divides two values together to compute their quotient.</summary>
+    /// <param name="left">The value which <paramref name="right" /> divides.</param>
+    /// <param name="right">The value which divides <paramref name="left" />.</param>
+    /// <returns>The quotient of <paramref name="left" /> divided-by <paramref name="right" />.</returns>
+    /// <exception cref="OverflowException">The quotient of <paramref name="left" /> divided-by <paramref name="right" /> is not representable by <typeparamref name="TVector" />.</exception>
+    static virtual TVector operator checked /(TVector left, TScalar right) => left / right;
+
+    /// <summary>Multiplies two values together to compute their product.</summary>
+    /// <param name="left">The value which <paramref name="right" /> multiplies.</param>
+    /// <param name="right">The value which multiplies <paramref name="left" />.</param>
+    /// <returns>The product of <paramref name="left" /> multiplied-by <paramref name="right" />.</returns>
+    static abstract TVector operator *(TVector left, TScalar right);
+
+    /// <summary>Multiplies two values together to compute their product.</summary>
+    /// <param name="left">The value which <paramref name="right" /> multiplies.</param>
+    /// <param name="right">The value which multiplies <paramref name="left" />.</param>
+    /// <returns>The product of <paramref name="left" /> multiplied-by <paramref name="right" />.</returns>
+    /// <exception cref="OverflowException">The product of <paramref name="left" /> multiplied-by <paramref name="right" /> is not representable by <typeparamref name="TVector" />.</exception>
+    static virtual TVector operator checked *(TVector left, TScalar right) => left * right;
+
+    /// <summary>Multiplies two values together to compute their product.</summary>
+    /// <param name="right">The value which <paramref name="left1" /> multiplies.</param>
+    /// <param name="left1">The value which multiplies <paramref name="right" />.</param>
+    /// <returns>The product of <paramref name="right" /> multiplied-by <paramref name="left1" />.</returns>
+    static virtual TVector operator *(TScalar left1, TVector right) => right * left1;
+
+    /// <summary>Multiplies two values together to compute their product.</summary>
+    /// <param name="right">The value which <paramref name="left" /> multiplies.</param>
+    /// <param name="left">The value which multiplies <paramref name="right" />.</param>
+    /// <returns>The product of <paramref name="right" /> multiplied-by <paramref name="left" />.</returns>
+    /// <exception cref="OverflowException">The product of <paramref name="right" /> multiplied-by <paramref name="left" /> is not representable by <typeparamref name="TVector" />.</exception>
+    static virtual TVector operator checked *(TScalar left, TVector right) => right * left;
```

Comment:
nit: this should be `checked(right * left)`, the other checked operators need to likewise reuse `checked` so that overflow checking can happen as expected.